File: appunvs/android/app/src/main/java/com/appunvs/runtime/ui/Badge.kt
// Badge — pill-shaped status label.  Tone semantics map to theme
// tokens, never raw colors at the call site.
package com.appunvs.runtime.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

import com.appunvs.runtime.theme.LocalAppColors
import com.appunvs.runtime.theme.Spacing

enum class BadgeTone { NEUTRAL, INFO, SUCCESS, WARNING, DANGER }

@Composable
fun AppBadge(
    label: String,
    tone: BadgeTone = BadgeTone.NEUTRAL,
    modifier: Modifier = Modifier,
) {
    val colors = LocalAppColors.current
    val (bg, fg) = when (tone) {
        BadgeTone.NEUTRAL -> colors.bgInput   to colors.textSecondary
        BadgeTone.INFO    -> colors.brandPale to colors.brandDark
        BadgeTone.SUCCESS -> colors.brandPale to colors.semanticSuccess
        BadgeTone.WARNING -> colors.bgInput   to colors.semanticWarning
        BadgeTone.DANGER  -> colors.bgInput   to colors.semanticDanger
    }
    Text(
        text = label,
        color = fg,
        fontSize = 13.sp,
        fontWeight = FontWeight.SemiBold,
        modifier = modifier
            .clip(RoundedCornerShape(percent = 50))
            .background(bg)
            .padding(horizontal = Spacing.s.dp, vertical = 2.dp),
    )
}

File: appunvs/ios/Runtime/UI/Badge.swift
// Badge — compact pill for status labels (draft / published / running /
// failed / etc).  Tone is the semantic role; the foreground/background
// pair is derived from theme tokens — never set explicit colors at the
// call site.
import SwiftUI

enum BadgeTone {
    case neutral, info, success, warning, danger
}

struct Badge: View {
    let label: String
    let tone: BadgeTone

    init(_ label: String, tone: BadgeTone = .neutral) {
        self.label = label
        self.tone = tone
    }

    var body: some View {
        Text(label)
            .font(.caption.weight(.semibold))
            .foregroundStyle(foreground)
            .padding(.horizontal, Spacing.s)
            .padding(.vertical, 2)
            .background(
                Capsule().fill(background)
            )
    }

    private var foreground: Color {
        switch tone {
        case .neutral: return Theme.textSecondary.color
        case .info:    return Theme.brandDark.color
        case .success: return Theme.semanticSuccess.color
        case .warning: return Theme.semanticWarning.color
        case .danger:  return Theme.semanticDanger.color
        }
    }

    private var background: Color {
        switch tone {
        case .neutral: return Theme.bgInput.color
        case .info, .success: return Theme.brandPale.color
        case .warning, .danger: return Theme.bgInput.color
        }
    }
}

#Preview {
    HStack(spacing: 8) {
        Badge("draft", tone: .warning)
        Badge("published", tone: .success)
        Badge("running", tone: .info)
        Badge("failed", tone: .danger)
        Badge("archived", tone: .neutral)
    }
    .padding()
}
